### ISSUE TITLE
`@stylistic/unit-case` names a fabricated unit for a word holding a multiplication, and reports only its first part

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#297]
Found while reviewing the branch that closes [#234](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/234). Reproduces on `main` before that branch as it does after it.

### What minimal example or steps are needed to reproduce the bug?

A word holding more than one dimension, which a `*` between them makes:

```css
a { b: 10PX*2REM*3EM; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "lower"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json
```

### What did you expect to happen?

Three warnings, one per unit, each naming the unit it is about: `PX`, `REM`, `EM`.

### What actually happened?

```
1:15-1:21  Expected "PX" to be "px"
1:10-1:21  Expected "PX*2REM*3EM" to be "px*2rem*3em"
```

Three faults in one output:

- `PX*2REM*3EM` is named as a unit, and no such unit exists — a message no user can act on;
- the same `PX` draws two warnings, one from the whole word and one from the part;
- `REM` and `EM` are never named at all. The scan of the parts is a `.some()`, so it stops at the first part that has something wrong with it.

The fix does reach every unit of the word, since the write recases the whole of it, so this costs the reader rather than the file.

### What is the cause?

`lib/rules/unit-case/index.js` walks the parts of such a word and the word itself:

```js
if (value.includes(`*`)) {
	value.split(`*`).some((val) => processValue({ … }))
}

if (processValue(valueNode)) …
```

The second call is what writes, and it reads the whole word as one dimension, `valueParser.unit` handing back everything behind the first number. So the word is measured twice under two different readings, and only one of them is a unit.

### Do you have a proposal to fix the bug?

Report from the parts alone and write from the word: walk every part rather than stopping at the first, take the whole-word reading out of the report, and give each part the position it stands at. Where the parts report is [#296](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/296), a separate defect: the positions such a node carries describe no text at all.
